Fix AutogradMonkeyPatch typo, enable replacements

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89457
* #89802
* __->__ #89520

- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

wconstab committed:
Fix AutogradMonkeyPatch typo, enable replacements

- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

[ghstack-poisoned]

wconstab commented on torch/_inductor/overrides.py:
```diff
@@ -28,7 +28,7 @@ class AutogradMonkeypatch(TorchFunctionMode):
         if not kwargs:
             kwargs = {}
-        if func is replacements:
+        if func in replacements:
```
@jansel do you think this codepath should be needed?  or if things were working ok before should we just delete AutogradMonkeyPatch?

jansel replied:
Ideally no, in practice maybe?

The dropout replacements was happening pretty inconsistently, so this was an (obviously broken) attempt to get them to work in a more robust way.  I think the bigger fix is to allow for pre-autograd replacements in AotAutograd.

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab committed:
Update on "Fix AutogradMonkeyPatch typo, enable replacements"


- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge

wconstab commented:
oops this got abandoned but someone else fixed it!
https://github.com/pytorch/pytorch/pull/90901
thanks!
